### PR TITLE
20490-NewValueHolder-substituable-for-ValueHolder

### PR DESCRIPTION
**Pull request by Ducasse**
https://pharo.fogbugz.com/f/cases/20490/NewValueHolder-substituable-for-ValueHolder

### CODE REVIEW

File: src/Morphic-Examples.package/WidgetExamples.class/class/colorControls.st [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+									help: 'This is a saturation-volume selector') minHeight: 80).
+			'HSV Color Selector'->((dialog newHSVSelector: Color green
+									help: 'This is a hue-saturation-volume selector') minHeight: 80; minWidth: 120)}.
+		dialog newVerticalSeparator.
+		dialog newTitle: 'HSVA Color Selector' for:
+			((dialog newHSVASelector: Color cyan
+					help: 'This is a hue-saturation-volume-alpha selector')
+					minHeight: 184; minWidth: 184)});
+		model: nil.
+	builder openModal: dialog
```
Comment (pavel-krivanek) [inline src/Morphic-Examples.package/WidgetExamples.class/class/colorControls.st]: should not the sample instance return the instance?

File: src/Morphic-Examples.package/WidgetExamples.class/class/colorControls.st [hunk 2]
```diff
@@ -0,0 +1,25 @@
+examples
+colorControls
+	"Open a window with examples of each of the color controls."
+	"self exampleColorControls"
```
Comment (pavel-krivanek) [inline src/Morphic-Examples.package/WidgetExamples.class/class/colorControls.st]: wrong comment

File: src/Morphic-Examples.package/WidgetExamples.class/class/exampleBasicControls.st [hunk 3]
```diff
@@ -2,7 +2,7 @@ examples
 exampleBasicControls
 	"Open a window with examples of each of the basic controls."
 	"self exampleBasicControls"
-	
+	<sampleInstance>
 	|dialog builder radioModel treeModel|
 	builder := self exampleBuilder.
 	dialog := (builder newPluggableDialogWindow: 'Example basic controls') useDefaultOKButton.
```
Comment (pavel-krivanek) [inline src/Morphic-Examples.package/WidgetExamples.class/class/exampleBasicControls.st]: again, sample instance not returned

File: src/Morphic-Examples.package/WidgetExamples.class/class/exampleWindowWithToolbars.st [hunk 4]
```diff
@@ -56,11 +56,11 @@ exampleWindowWithToolbars
...
+	holder := NewValueHolder new.
 	text := win
 		newTextEditorFor: holder
-		getText: #contents
-		setText: #contents:.
+		getText: #value
+		setText: #value:.
 	dock := win newToolDockingBar.
 	dock
 		add: 'File' translated
```
Comment (pavel-krivanek) [inline src/Morphic-Examples.package/WidgetExamples.class/class/exampleWindowWithToolbars.st]: sample instance not returned

File: src/NewValueHolder.package/NewValueHolder.class/properties.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
 	"commentStamp" : "PeterUhnak 3/5/2016 11:16",
-	"super" : "Object",
```
Comment (peteruhnak) [inline src/NewValueHolder.package/NewValueHolder.class/properties.json]: Why is NewValueHolder now subclass of Model?

Comment (Ducasse) [reply src/NewValueHolder.package/NewValueHolder.class/properties.json]: May be this is the wrong analysis from me.
In fact PluggableButtonMorph requires something that can accept addDependent:
So when I converted the Value to NewValueHolder I found this.

**Comment by Ducasse** [comment]
Pavel I will have a look and address your point. 
May be I should use script instead of sampleInstance because returning an instance is not interesting

File: src/NewValueHolder.package/NewValueHolder.class/properties.json (hunk 5 above)
Comment (estebanlm) [reply src/NewValueHolder.package/NewValueHolder.class/properties.json]: yes, and Model is the right place to put that stuff... not object (we need to remove that mechanism from Object, btw... but is hard :P)

**Pull request merged**